include/nuttx/fs/fs.h

```diff
@@ -497,6 +546,8 @@ struct file_struct
   FAR struct file_struct *fs_next;      /* Pointer to next file stream */
   rmutex_t                fs_lock;      /* Recursive lock */
   int                     fs_fd;        /* File descriptor associated with stream */
```

Comment:
fs_fd should union with fs_cookie.cookie

Reply:
This should not be needed since I would pass the entire stream to internal callback.

Reply:
but the current design waste the memory:

1. fs_fd dup with fs_cookie.cookie
2. fs_callbacks dup with fs_cookie.cookie_io

total bytes equal to 20(32bit) or 36(64bit).

Reply:
I removed  `fs_fd` entirely as it is not needed anymore. Instead all file descriptors are passed through `fs_cookie` (which is used for user cookie if fopencookie is used). `fs_callbacks` removed.